controllers/etcdpeer_controller.go

```diff
@@ -103,6 +103,88 @@ func defineReplicaSet(peer etcdv1alpha1.EtcdPeer, log logr.Logger) appsv1.Replic
 		peerLabel:    peer.Name,
 	}
 
+	etcdContainer := corev1.Container{
+		Name:  appName,
+		Image: etcdImage,
+		Env: []corev1.EnvVar{
+			{
+				Name:  etcdenvvar.InitialCluster,
+				Value: staticBootstrapInitialCluster(*peer.Spec.Bootstrap.Static),
+			},
+			{
+				Name:  etcdenvvar.Name,
+				Value: peer.Name,
+			},
+			{
+				Name:  etcdenvvar.InitialClusterToken,
+				Value: peer.Spec.ClusterName,
+			},
+			{
+				Name:  etcdenvvar.InitialAdvertisePeerURLs,
+				Value: advertiseURL(peer, etcdPeerPort).String(),
+			},
+			{
+				Name:  etcdenvvar.AdvertiseClientURLs,
+				Value: advertiseURL(peer, etcdClientPort).String(),
+			},
+			{
+				Name:  etcdenvvar.ListenPeerURLs,
+				Value: bindAllAddress(etcdPeerPort).String(),
+			},
+			{
+				Name:  etcdenvvar.ListenClientURLs,
+				Value: bindAllAddress(etcdClientPort).String(),
+			},
+			{
+				Name:  etcdenvvar.InitialClusterState,
+				Value: clusterStateValue(peer.Spec.Bootstrap.InitialClusterState),
+			},
+			{
+				Name:  etcdenvvar.DataDir,
+				Value: etcdDataMountPath,
+			},
+		},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "etcd-client",
+				ContainerPort: etcdClientPort,
+			},
+			{
+				Name:          "etcd-peer",
+				ContainerPort: etcdPeerPort,
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "etcd-data",
+				MountPath: etcdDataMountPath,
+			},
+		},
+	}
+	if peer.Spec.PodTemplate != nil {
+		if peer.Spec.PodTemplate.Resources != nil {
+			etcdContainer.Resources = *peer.Spec.PodTemplate.Resources
+			cpuLimit := etcdContainer.Resources.Limits.Cpu()
+			// GOMAXPROCS defaults to the number of CPUs on the Kubelet host
+			// which may be much higher than the requests and limits defined for the pod,
+			// See https://github.com/golang/go/issues/33803
+			// If a CPU limit has been set, then set GOMAXPROCS to an integer
+			// which between 1 and floor(cpuLimit).
+			// Etcd might one day set it's own GOMAXPROCS based on CPU quota:
+			// See: https://github.com/etcd-io/etcd/issues/11508
+			goMaxProcs := cpuLimit.MilliValue() / 1000
```

Comment:
In the event that the CPU limit is not set, default to not setting `GOMAXPROCS` as per comment?

Reply:
I've updated it now.
Only set GOMAXPROCS if CPU limit is > 0.

With some unit tests.